lte/gateway/c/session_manager/LocalEnforcer.cpp

```diff
@@ -1150,31 +1164,31 @@ void LocalEnforcer::update_charging_credits(
     for (const auto& session : it->second) {
       std::string sid                = session->get_session_id();
       SessionStateUpdateCriteria& uc = session_update[imsi][sid];
-      bool is_redirected =
-          session->is_credit_state_redirected(CreditKey(credit_update_resp));
+      bool is_final_action_state =
+          session->is_credit_in_final_unit_state(CreditKey(credit_update_resp));
       session->receive_charging_credit(credit_update_resp, uc);
-
       session->set_tgpp_context(credit_update_resp.tgpp_ctx(), uc);
       SessionState::SessionInfo info;
 
-      if (is_redirected) {
+      if (is_final_action_state) {
```

Comment:
Can you add a comment indicating that we are undoing FUA since it has received a new credit?

Reply:
Maybe, if possible, it would be better to put it in inside a function `cancelling_final_unit_action`

Note that the compiler will inline those functions. So small functions that are called not that often has no impact on performance and it improves readability.